src/lib/math.ts

```diff
@@ -3,15 +3,18 @@ import BigNumber from 'bignumber.js'
 import type { BN } from './bignumber/bignumber'
 import { bn, bnOrZero } from './bignumber/bignumber'
 
+// Converts from base unit to a precision/ish number
+// If no displayDecimals are provided, it will use the full precision of the number being converted
+// If displayDecimals are provided, it will use that precision, to return e.g a human, precision, or number stripped to any arbitrary decimal places
 export const fromBaseUnit = (
   value: BigNumber.Value,
-  decimals: number,
-  displayDecimals = 6,
+  precision: number,
+  displayDecimals?: number,
 ): string => {
   return bnOrZero(value)
-    .div(`1e+${decimals}`)
-    .decimalPlaces(displayDecimals, BigNumber.ROUND_DOWN)
-    .toString()
+    .div(bn(10).pow(precision))
+    .decimalPlaces(displayDecimals ?? precision, BigNumber.ROUND_DOWN)
```

Comment:
That `BigNumber.ROUND_DOWN` may also have been causing quite a few issues but prefer not to tackle it all in the same PR, changing it is quite high risk on an already high risk PR